Sources/XMLCoding/Decoder/XMLDecoder.swift

```diff
@@ -216,7 +246,45 @@ internal class _XMLDecoder : Decoder {
                                                                     debugDescription: "Cannot get keyed decoding container -- found null value instead."))
         }
         
-        guard let topContainer = self.storage.topContainer as? [String : Any] else {
+        let topContainer: [String : Any]
+        // if this is a dictionary
+        if let currentContainer = self.storage.topContainer as? [String : Any] {
+            
+            // if we are combining collapsing lists and maps
+            if case let .collapseListUsingItemTag(itemTag) = options.listDecodingStrategy,
+                case let .collapseMapUsingTags(keyTag: keyTag, valueTag: valueTag) = options.mapDecodingStrategy,
+                let itemList = currentContainer[itemTag] as? [Any] {
+                    var newContainer: [String: Any] = [:]
+                
+                    // construct a dictionary from each entry and the key and value tags
+                    itemList.forEach { entry in
```

Comment:
this and 278 looks like it could be put in a method to avoid repetitive code possibly?